src/mono/mono/utils/mono-context.h

```diff
@@ -45,9 +45,15 @@ typedef struct _libc_xmmreg MonoContextSimdReg;
 typedef __m128d MonoContextSimdReg;
 #endif
 #elif defined(TARGET_ARM64)
+#if defined(HOST_WIN32)
+#define MONO_HAVE_SIMD_REG
```

Comment:
This confused me a little at first, but needed since msvc doesn't implement __uint128_t so we need an similar type. Will it however be needed when you build a cross compiler, MONO_CROSS_COMPILE?

Reply:
Yeah I needed something that wasn't __uint128_t but kinda tasted like it'd fit? 🤷‍♂️ 

I'm not sure if it's really _needed_, but the existing code in TARGET_ARM64 wasn't compiling

Reply:
looks like it still compiles "stubs" that will assert on call for cross compiler, so even if it won't be used, it will still be needed to compile functions like mono_sigctx_to_monoctx that has MonoContext as parameter that in turn is depending on MonoContextSimdReg.

Maybe add a comment that we will need a replacement type for __uint128_t since it is not implemented under msvc (looking at the docs for __m128d it says its not supported on ARM processors and the emmintrin.h is specific to X86/x64 targets).

An alternative, do what emmintrin.h does:

```
typedef struct __declspec(intrin_type) __declspec(align(16)) __m128d {
    double              m128d_f64[2];
} __m128d;
``` 

but just adding a comment will why its needed will probably do as well.